docs/fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md

```diff
@@ -119,6 +132,123 @@ using var fileStream = new FileStream("MyAssembly.exe", FileMode.Create, FileAcc
+    MethodDefinitionHandle entryPointHandle = MetadataTokens.MethodDefinitionHandle(entryPoint.MetadataToken);
+    DebugDirectoryBuilder debugDirectoryBuilder = GeneratePDB(pdbBuilder, metadataBuilder.GetRowCounts(), entryPointHandle);
+
+    ManagedPEBuilder peBuilder = new ManagedPEBuilder(
+                    header: new PEHeaderBuilder(imageCharacteristics: Characteristics.ExecutableImage, subsystem: Subsystem.WindowsCui),
+                    metadataRootBuilder: new MetadataRootBuilder(metadataBuilder),
+                    ilStream: ilStream,
+                    debugDirectoryBuilder: debugDirectoryBuilder,
+                    entryPoint: entryPointHandle);
+
+    BlobBuilder peBlob = new BlobBuilder();
+    peBuilder.Serialize(peBlob);
+
+    using var fileStream = new FileStream("MyAssembly.exe", FileMode.Create, FileAccess.Write);
+    peBlob.WriteContentTo(fileStream);
+}
+
+static DebugDirectoryBuilder GeneratePDB(MetadataBuilder pdbBuilder, ImmutableArray<int> rowCounts, MethodDefinitionHandle entryPointHandle)
+{
+    BlobBuilder portablePdbBlob = new BlobBuilder();
+    PortablePdbBuilder portablePdbBuilder = new PortablePdbBuilder(pdbBuilder, rowCounts, entryPointHandle);
+    BlobContentId pdbContentId = portablePdbBuilder.Serialize(portablePdbBlob);
+    // In case saving PDB to a file
+    using FileStream fileStream = new FileStream("MyAssemblyEmbeddedSource.pdb", FileMode.Create, FileAccess.Write);
+    portablePdbBlob.WriteContentTo(fileStream);
+
+    DebugDirectoryBuilder debugDirectoryBuilder = new DebugDirectoryBuilder();
+    debugDirectoryBuilder.AddCodeViewEntry("MyAssemblyEmbeddedSource.pdb", pdbContentId, portablePdbBuilder.FormatVersion);
+    // In case embedded in PE:
+    // debugDirectoryBuilder.AddEmbeddedPortablePdbEntry(portablePdbBlob, portablePdbBuilder.FormatVersion);
+    return debugDirectoryBuilder;
+}
+```
+
+Further user could add CustomDebugInformation by calling the AddCustomDebugInformation method from the `pdbBuilder` instance to add source embedding and source indexing etc. advanced PDB info.
+
+```csharp
+private static void EmbedSource(MetadataBuilder pdbBuilder)
+{
+    byte[] sourceBytes = File.ReadAllBytes("MySourceFile2.cs");
+    BlobBuilder sourceBlob = new BlobBuilder();
+    sourceBlob.WriteBytes(sourceBytes);
+    pdbBuilder.AddCustomDebugInformation(MetadataTokens.DocumentHandle(1),
+        pdbBuilder.GetOrAddGuid(new Guid("0E8A571B-6926-466E-B4AD-8AB04611F5FE")), pdbBuilder.GetOrAddBlob(sourceBlob));
+}
+```
+
+### Adding resources with PersistedAssemblyBuilder
+
+With `MetadataBuilder.AddManifestResource(...)` you could add as many resources as needed, but streams have to be concatenated together into one `BlobBuilder` that you pass into `ManagedPEBuilder` argument.
+Following example shows how to create resources and attach it to the assembly created:
+
+```csharp
+public static void SetResource()
+{
+    PersistedAssemblyBuilder ab = CreateAssemblyAndItsContents();
```

Comment:
```suggestion
    PersistedAssemblyBuilder ab = new PersistedAssemblyBuilder(new AssemblyName("MyAssembly"), typeof(object).Assembly);
```
Would this work? It would be nice if people can copy&paste the examples and they compile.

Reply:
Had to define a module at least